app/gui2/src/components/GraphEditor/widgets/WidgetFunction.vue

```diff
@@ -46,6 +47,18 @@ const interpreted = computed(() => {
   return interpretCall(props.input.value, methodCallInfo.value == null)
 })
 
+const selfArgumentPreapplied = computed(() => {
+  const info = methodCallInfo.value
+  if (info?.staticallyApplied) return false
+  const analyzed = interpreted.value
+  if (analyzed.kind !== 'prefix') return false
+  const subject = getAccessOprSubject(analyzed.func)
```

Comment:
In those cases, we don't actually want to strip `self`. Instead, we rely on `noArgsCall.notAppliedArguments` to tell us if it is applied or not. Notably, that reliably works ONLY in cases where the first argument is a function variable (i.e. assigned in other node). Also, this logic does occasionally cause small glitching of the argument lists for those "function in variable" application right after editing, since updating `notAppliedArguments` relies on expression update. That of course has some latency. But I've managed to mitigate that whenever possible. At worst, in some rare edge cases the arguments will temporarily disappear after disconnect.